### PR TITLE
Implement obfuscation to protect Transifex credentials

### PR DESCRIPTION
**Pull request by mauricecarrier7**
**What's this do?**
Integrates obfuscator to encode/decode transifex credentials

**Why are we doing this? (w/ Notion link if applicable)**
https://www.notion.so/lyrasis/Transifex-Integrate-and-Obfuscate-APIToken-and-Secret-8457dda7083a4b6192d63ed4c1b78d37?d=3402ac26a1c14be98922d7da94d90565#501ebb27dcf24047bfc51a1ab964ee39

**How should this be tested? / Do these changes have associated tests?**
This is unit tested

**Dependencies for merging? Releasing to production?**
N/A

**Does this include changes that require a new Palace build for QA?**
No

**Has the application documentation been updated for these changes?**
N/A
**Did someone actually run this code to verify it works?**
@mauricecarrier7

### CODE REVIEW

File: Palace/Security/Obfuscator.swift [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        let cipher = [UInt8](self.salt.utf8)
+        let length = cipher.count
+        
+        var encrypted = [UInt8]()
+        
+        for t in text.enumerated() {
+            encrypted.append(t.element ^ cipher[t.offset % length])
+        }
+        
+        #if DEVELOPMENT
```
Comment (vladimirfedorov) [inline Palace/Security/Obfuscator.swift:47]: Where is `DEVELOPMENT` defined? I couldn't find this constant in the project, but we have `DEBUG` for Debug mode

File: Palace/Utilities/Localization/TransifexManager.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+//  Palace
+//
+//  Created by Maurice Carrier on 9/30/22.
+//  Copyright © 2022 The Palace Project. All rights reserved.
+//
+
+import Foundation
+
+struct TransifexManager {
+  static let token: [UInt8] = [101, 127, 97, 112, 66, 71, 38, 7, 15, 7, 80, 2, 76, 92, 125, 103, 42, 86,
```
Comment (vladimirfedorov) [inline Palace/Utilities/Localization/TransifexManager.swift:12]: Is it ok that we keep token here and not in secrets?

File: Palace/Security/Obfuscator.swift (hunk 1 above)
Comment (mauricecarrier7) [reply Palace/Security/Obfuscator.swift:47]: This is borrowed code, this should be removed.

File: Palace/Utilities/Localization/TransifexManager.swift (hunk 2 above)
Comment (mauricecarrier7) [reply Palace/Utilities/Localization/TransifexManager.swift:12]: Ah, completely missed TPPSecrets, where is the encoding function, I can reuse the salt to encode the token

Comment (mauricecarrier7) [reply Palace/Utilities/Localization/TransifexManager.swift:12]: Ah, I recreated the existing obfuscator. Will update

**Pull request closed without merging**